spec_verifier/mlperf_spec_verifier.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+import re
+import json
+from enum import Enum
+
+# Parser for command line
+parser = argparse.ArgumentParser(description='Input log, model, and mode.')
+parser.add_argument('--log', action='store', type=str,
+                    dest='log_filename', help='Path of log file')
+parser.add_argument('--spec', action='store', type=str,
+                    dest='spec_filename', help='Path of spec JSON file')
+parser.add_argument('--model', action='store', type=str, dest='model_name',
+                    help='Model name you are testing (e.g. Resnet50-v1.5)')
+
+args = parser.parse_args()
+
+field_compliance = {}
+
+class ExitCode (Enum):
+    COMPLIANT = "COMPLIANT"
+    SETTING_ERROR = "SETTING_ERROR"
+    SCENARIO_ERROR = "SCENARIO_ERROR"
+
+# Scrape lines from txt file
+with open(args.log_filename) as logfile:
+    log_lines = logfile.readlines()
+
+log_lines = [line.strip() for line in log_lines]
+
+# Import JSON file
+with open(args.spec_filename) as json_file:
+    json_dict = json.load(json_file)
+
+def find_string(iter_list, substring):
+    """Return index if a specific substring is inside each string in a list."""
+    for i, s in enumerate(iter_list):
+        if substring in s:
+            return i
+
+
+effective_string = 'Effective Settings:'
+requested_string = 'Requested Settings:'
+
+effective_index = find_string(log_lines, effective_string)
+requested_index = find_string(log_lines, requested_string)
+
+effective_settings = log_lines[effective_index+1:requested_index-1]
+effective_dict = {}
+
+
+# Find every ":" in each line
+def find_char(character, string):
+    """Returns index for every time a character is found in a string."""
+    char_counter = []
+    for m in re.finditer(character, string):
+        char_counter.append(m.start())
+    return char_counter
+
+
+def slice_char(string, character, index):
+    """Slices a string to remove everything before a specific character."""
+    return string[find_char(character, string)[index] + 2:]

```

Comment:
Why +2? Can you add a constant or comment? Ditto for other constants in the code.

Reply:
Done. Let me know if more clarification is needed on the constants.